test/integration/fake_upstream.cc

```diff
@@ -407,6 +407,16 @@ std::string FakeRawConnection::waitForData(uint64_t num_bytes) {
   return data_;
 }
 
+std::string FakeRawConnection::waitForData(const char* data, bool exact_match) {
```

Comment:
What about this more-generalized version? I wrote this for an internal test, and have been meaning to upstream it, and now seems like a good time.

```
std::string FakeRawConnection::waitForData(const std::function<bool(const std::string&)>& filter) {
  Thread::LockGuard lock(lock_);
  while (!filter(data_)) {
    ENVOY_LOG(debug, "waiting for data");
    connection_event_.wait(lock_); // Safe since CondVar::wait won't throw.
  }
  return data_;
}
```